### PR TITLE
wait_for_up() in tests running guest commands

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

- Insert `wait_for_up()` before running guest commands.

## Reason

`wait_for_up()` shows debugging information if the guest is not up and running.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this
  PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/functional/test_api.py [hunk 1]
```diff
@@ -1167,8 +1167,7 @@ def test_get_full_config_after_restoring_snapshot(microvm_factory, uvm_nano):
     snapshot = uvm_nano.snapshot_full()
     uvm2 = microvm_factory.build()
     uvm2.spawn()
-    uvm2.restore_from_snapshot(snapshot)
-    uvm2.resume()
+    uvm2.restore_from_snapshot(snapshot, resume=True)
```
[the remark this answers is not in the document]
Comment (zulinx86) [reply tests/integration_tests/functional/test_api.py:1170]: indeed. I'll cover both cases!

File: tests/integration_tests/functional/test_snapshot_basic.py [hunk 2]
```diff
@@ -51,6 +51,45 @@ def _get_guest_drive_size(ssh_connection, guest_dev_name="/dev/vdb"):
     return lines[1].strip()
 
 
+def test_resume_after_restoration(uvm_nano, microvm_factory):
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_snapshot_basic.py:54]: nit: this can go in a separate commit

**Review by kalyazin — APPROVED** [review]

**Review by bchalios — COMMENTED** [review]
Same comment as Nikita's about splitting out the new tests in their own commit. Other than that LGTM
